android/app/src/main/kotlin/com/therealansh/iiit_sonepat_stable/MainActivity.kt

```
@@ -1,6 +1,128 @@
+
+    private val broadcastReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            onBroadcastReceived(intent)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Initialize default options for Jitsi Meet conferences.
+        val serverURL: URL = try {
+            // When using JaaS, replace "https://meet.jit.si" with the proper serverURL
+            URL("https://meet.jit.si")
+        } catch (e: MalformedURLException) {
+            e.printStackTrace()
+            throw RuntimeException("Invalid server URL!")
+        }
+        val defaultOptions = JitsiMeetConferenceOptions.Builder()
+                .setServerURL(serverURL)
+                // When using JaaS, set the obtained JWT here
+                //.setToken("MyJWT")
+                // Different features flags can be set
+                //.setFeatureFlag("toolbox.enabled", false)
+                //.setFeatureFlag("filmstrip.enabled", false)
+                .setWelcomePageEnabled(false)
+                .build()
+        JitsiMeet.setDefaultConferenceOptions(defaultOptions)
+
+        registerForBroadcastMessages()
+    }
+
+    override fun onDestroy() {
+        LocalBroadcastManager.getInstance(this).unregisterReceiver(broadcastReceiver)
+        super.onDestroy()
+    }
+
+    var usertext:String?=""//For storing UserName
+
+    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
+        super.configureFlutterEngine(flutterEngine)
+        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler { call, result ->
+
+            if(call.method == "Meeting started")
+                usertext=call.argument<String>("text")
+            onButtonClick(usertext)
+
+        }
+    }
+
+    fun onButtonClick(text:String?) {
```

we can handle error here too, we could raise an Exception here in case a user is unable to join the call. This exception can be showed at the frontend. Read about ``` try{} catch(err) {}``` block statement to raise an exception and we could show a dialog box in the frontend if we catch an error